models/ed/R/met2model.ED2.R

```diff
@@ -176,16 +174,32 @@ met2model.ED2 <- function(in.path, in.prefix, outfolder, start_date, end_date, l
     useCO2 <- is.numeric(CO2)
 
     ## convert time to seconds
-    sec <- udunits2::ud.convert(sec, unlist(strsplit(nc$dim$time$units, " "))[1], "seconds")
+    sec <- udunits2::ud.convert(tdays, unlist(strsplit(nc$dim$time$units, " "))[1], "seconds")
 
     ncdf4::nc_close(nc)
 
-    dt <- PEcAn.utils::seconds_in_year(year, leap_year) / length(sec)
+    # `dt` is the met product time step in seconds. We calculate it here as
+    # `sec[i] - sec[i-1]` for all `[i]`. For a properly formatted product, the
+    # timesteps should be regular, so there is a single, constant difference. If
+    # that's not the case, we throw an informative error.
+    #
+    # `drop` here simplifies length-1 arrays to vectors. Without it, R will
+    # later throw an error about "non-conformable arrays" when trying to add a
+    # length-1 array to a vector.
+    dt <- drop(unique(diff(sec)))
+    if (length(dt) > 1) {
+      PEcAn.logger::logger.severe(paste0(
+        "Time step (`dt`) is not uniform! Identified ",
+        length(dt), " unique time steps. ",
+        "`head(dt)` (in seconds): ",
+        paste(head(dt), collapse = ", ")
+      ))
+    }
 
     toff <- -as.numeric(lst) * 3600 / dt
 
     ## buffer to get to GMT
-    slen <- seq_along(SW)
+    slen <- seq_along(sec)
```

Comment:
All of these vectors pretty much have to be the length by definition (unless the met file is mis-specified), so the exact choice of variable here shouldn't matter. It seemed a bit clearer to me to use the time variable rather than arbitrarily choosing one of the actual met variables (which, in theory, could be multidimensional, though that's not actually the case).